Authentication/SignUp/ViewModel/SignupViewModel.swift

```diff
@@ -139,34 +138,34 @@ public final class SignupViewModel<User, SessionService: SessionServiceType wher
 
 private extension SignupViewModel {
     
-    private func initializeSignUpAction() -> Action<AnyObject, User, SessionServiceError> {
+    func initializeSignUpAction() -> Action<(), User, SessionServiceError> {
         return Action(enabledIf: self._credentialsAreValid) { [unowned self] _ in
             if let email = Email(raw: self.email.value) {
-                let username: String? = self.usernameEnabled ? self.username.value : .None
+                let username: String? = self.usernameEnabled ? self.username.value : .none
                 return self._sessionService.signUp(username, email: email, password: self.password.value)
             } else {
                 // It will never enter here, since sign up action is only enabled when email is a valid email.
-                return SignalProducer(error: .InvalidSignUpCredentials(.None)).observeOn(UIScheduler())
+                return SignalProducer(error: .invalidSignUpCredentials(.none)).observe(on: UIScheduler())
             }
         }
     }
     
-    private func initializeTogglePasswordVisibilityAction() -> Action<AnyObject, Bool, NoError> {
+    func initializeTogglePasswordVisibilityAction() -> Action<(), Bool, NoError> {
         return Action { [unowned self] _ in
             self.passwordVisible.value = !self.passwordVisible.value
-            return SignalProducer(value: self.passwordVisible.value).observeOn(UIScheduler())
+            return SignalProducer(value: self.passwordVisible.value).observe(on: UIScheduler())
         }
     }
     
-    private func initializeTogglePasswordConfirmationVisibilityAction() -> Action<AnyObject, Bool, NoError> {
+    func initializeTogglePasswordConfirmationVisibilityAction() -> Action<(), Bool, NoError> {
         return Action { [unowned self] _ in
             self.passwordConfirmationVisible.value = !self.passwordConfirmationVisible.value
-            return SignalProducer(value: self.passwordConfirmationVisible.value).observeOn(UIScheduler())
+            return SignalProducer(value: self.passwordConfirmationVisible.value).observe(on: UIScheduler())
         }
     }
     
 }
 
-private func getPasswordConfirmValidationResultFromEquality(equals: Bool) -> ValidationResult {
+private func getPasswordConfirmValidationResultFromEquality(_ equals: Bool) -> ValidationResult {
```

Comment:
This works in `private` and not `fileprivate` ? I don't think I get the difference so much then. Although this isn't inside a class so as to say it's private to that type.